tests/test_response_required_logic.py

```diff
@@ -185,21 +187,52 @@ def test_is_provider_using_response():
 
 def test_is_callback_using_response():
     spider = MySpider()
-    assert is_callback_requiring_scrapy_response(spider.parse) is True
-    assert is_callback_requiring_scrapy_response(spider.parse2) is True
-    assert is_callback_requiring_scrapy_response(spider.parse3) is False
-    assert is_callback_requiring_scrapy_response(spider.parse4) is False
-    assert is_callback_requiring_scrapy_response(spider.parse5) is True
-    assert is_callback_requiring_scrapy_response(spider.parse6) is False
-    assert is_callback_requiring_scrapy_response(spider.parse7) is True
-    assert is_callback_requiring_scrapy_response(spider.parse8) is False
-    assert is_callback_requiring_scrapy_response(spider.parse9) is True
-    assert is_callback_requiring_scrapy_response(spider.parse10) is False
-    assert is_callback_requiring_scrapy_response(spider.parse11) is True
-    assert is_callback_requiring_scrapy_response(spider.parse12) is True
+    mock_request = MagicMock()
+    assert is_callback_requiring_scrapy_response(spider.parse, mock_request) is True
+    assert is_callback_requiring_scrapy_response(spider.parse2, mock_request) is True
+    assert is_callback_requiring_scrapy_response(spider.parse3, mock_request) is False
+    assert is_callback_requiring_scrapy_response(spider.parse4, mock_request) is False
+    assert is_callback_requiring_scrapy_response(spider.parse5, mock_request) is True
+    assert is_callback_requiring_scrapy_response(spider.parse6, mock_request) is False
+    assert is_callback_requiring_scrapy_response(spider.parse7, mock_request) is True
+    assert is_callback_requiring_scrapy_response(spider.parse8, mock_request) is False
+    assert is_callback_requiring_scrapy_response(spider.parse9, mock_request) is True
+    assert is_callback_requiring_scrapy_response(spider.parse10, mock_request) is False
+    assert is_callback_requiring_scrapy_response(spider.parse11, mock_request) is True
+    assert is_callback_requiring_scrapy_response(spider.parse12, mock_request) is True
     # Callbacks created with the callback_for function won't make use of
     # the response, but their providers might use them.
-    assert is_callback_requiring_scrapy_response(spider.callback_for_parse) is False
+    assert (
+        is_callback_requiring_scrapy_response(spider.callback_for_parse, mock_request)
+        is False
+    )
+
+    # See: https://github.com/scrapinghub/scrapy-poet/issues/48
+    mock_request.callback = None
```

Comment:
can we use a real Request instad of mock?

Reply:
Good call! https://github.com/scrapinghub/scrapy-poet/pull/106/commits/e353f81df552fea79d7ec12a394c9b6509bc2db2